### PR TITLE
Setup reporting for missing CSS file

### PR DESCRIPTION
**Pull request by prakhargurunani**
fixes: #388 

Returns missing CSS files in `OSError` according to the order they were passed.

### CODE REVIEW

**Review by gnikonorov — CHANGES_REQUESTED** [review]
Thanks for raising this @prakhargurunani!

Can you please also add:
- A [changelog entry](https://github.com/pytest-dev/pytest-html/blob/master/CHANGES.rst)
- Some [tests for this change](https://github.com/pytest-dev/pytest-html/blob/master/testing/test_pytest_html.py)

**Comment by prakhargurunani** [comment]
@gnikonorov I have only changed the way of reporting the missing CSS files. I can't see any new test cases for this change. If any, Please let me know.

**Comment by gnikonorov** [comment]
> @gnikonorov I have only changed the way of reporting the missing CSS files. I can't see any new test cases for this change. If any, Please let me know.

In the [run](https://github.com/pytest-dev/pytest-html/blob/3f6368693f4316372c0b7113ecf0ccd1300e7a05/testing/test_pytest_html.py#L39) method you can pass a variable list of arguments via `*args`. You could pass a list of css files that don't exist and confirm behavior via the [css flag](https://github.com/pytest-dev/pytest-html/blob/3f6368693f4316372c0b7113ecf0ccd1300e7a05/src/pytest_html/plugin.py#L66)

**Comment by prakhargurunani** [comment]
Thanks @gnikonorov !

File: src/pytest_html/plugin.py
Comment (BeyondEvil) [inline src/pytest_html/plugin.py:99]:
```suggestion
        missing_css_files = []
        for csspath in config.getoption("css"):
            if not os.path.exists(csspath):
                missing_css_files.append(csspath)

        if missing_css_files:
            raise IOError(
                f"Missing CSS file{'s' if len(missing_css_files) > 1 else ''}: '{', '.join(missing_css_files)}'"
            )
```

**Review by BeyondEvil — CHANGES_REQUESTED** [review]

File: src/pytest_html/plugin.py
Comment (prakhargurunani) [reply src/pytest_html/plugin.py:99]: Thanks @BeyondEvil !

**Comment by prakhargurunani** [comment]
> In the [run](https://github.com/pytest-dev/pytest-html/blob/3f6368693f4316372c0b7113ecf0ccd1300e7a05/testing/test_pytest_html.py#L39) method you can pass a variable list of arguments via `*args`. You could pass a list of css files that don't exist and confirm behavior via the [css flag](https://github.com/pytest-dev/pytest-html/blob/3f6368693f4316372c0b7113ecf0ccd1300e7a05/src/pytest_html/plugin.py#L66)

@gnikonorov Done.

**Review by BeyondEvil — DISMISSED** [review]
Thanks for this @prakhargurunani !

I'm satisfied but I'll let @gnikonorov have the last word (and/or merge). 👍

File: testing/test_pytest_html.py
Comment (gnikonorov) [inline testing/test_pytest_html.py:1015]: Can you add a test for where only part of the passed files are missing? You can use https://github.com/pytest-dev/pytest-html/blob/master/testing/test_pytest_html.py#L321 as an example for creating adhoc files in tests.

Also, please use [@pytest.mark.parametrize](https://docs.pytest.org/en/stable/parametrize.html) to combine your tests with the test above. It'll be a lot cleaner than having 2-3 separate tests for the same feature ( reporting of missing css files )

[Here](https://github.com/pytest-dev/pytest-html/blob/master/testing/test_pytest_html.py#L482) is an example of how we parametrize in this file

**Review by gnikonorov — CHANGES_REQUESTED** [review]
One last comment and then I think this is good to merge @prakhargurunani

File: testing/test_pytest_html.py
Comment (prakhargurunani) [reply testing/test_pytest_html.py]: @gnikonorov See https://github.com/pytest-dev/pytest-html/pull/390/commits/8ccc1045cfa80b463994feb5f418f96a613bcfd9
Have I used `@pytest.mark.parametrize` correctly ?

File: testing/test_pytest_html.py
Comment (gnikonorov) [inline testing/test_pytest_html.py:1024]: You can merge this test with the test above, just create the valid css file in the body of the test.

File: testing/test_pytest_html.py
Comment (gnikonorov) [inline testing/test_pytest_html.py:1015]: This is correct, but you could make it even more clean by parametrizing with a list of files instead of passing `file1` and `file2`. For example:
```
$ cat test_parametrize.py
import pytest

@pytest.mark.parametrize("arg", ["arg1", "arg2", ["arg3.a", "arg3.b"]])
def test_list(arg):
     assert arg is not None
$
```

In the example above the test is run 3 times. First with `"arg1"`, then `"arg2"`, and finally with the list `["arg3.a", "arg3.b"]`

This way your test can be expanded on in the future without changing its signature every time, and it becomes easier to reason about

**Review by gnikonorov — CHANGES_REQUESTED** [review]

File: testing/test_pytest_html.py
Comment (gnikonorov) [reply testing/test_pytest_html.py]: Almost @prakhargurunani! Please see my last review

**Comment by gnikonorov** [comment]
I'll take a look tonight @prakhargurunani! Sorry, I've been busy

**Review by gnikonorov — CHANGES_REQUESTED** [review]
Please fix the changelog. After that I'm fine with approving

File: testing/test_pytest_html.py
Comment (gnikonorov) [inline testing/test_pytest_html.py:1017]: Is this debug code? `files` should never be `None` based on your parametrization

File: testing/test_pytest_html.py
Comment (gnikonorov) [inline testing/test_pytest_html.py:1031]: Sorry I missed this, but this doesn't do what you expect. You aren't actually checking for `files[0]`. This just checks whether or not `files[0]` is truthy and executes `files[1] in result.stderr.str()` if it's not. For example:
```
>>> l = [1,2,3]
>>> assert 0, 2 in l
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError: True
>>> assert 44, 2 in l
```

**Review by gnikonorov — CHANGES_REQUESTED** [review]

**Review by gnikonorov — APPROVED** [review]
LGTM, thanks @prakhargurunani !

I'll leave the final review/merge up to @BeyondEvil

**Review by BeyondEvil — DISMISSED** [review]
Thanks for all the hard work @prakhargurunani !

This looks great!

Can you squash and rebase so we can merge?

**Comment by BeyondEvil** [comment]
@gnikonorov docs are failing, not sure what to do here? 🤔

**Comment by gnikonorov** [comment]
@BeyondEvil docs are failing because the RTD build was enabled before the PR for RTD was merged. Once https://github.com/pytest-dev/pytest-html/pull/402 goes in the issue will go away

**Review by BeyondEvil — APPROVED** [review]
Thanks everyone!

**Pull request merged**
